### PR TITLE
ENT-3642: move the crypto service builder method to node-api

### PR DESCRIPTION
**Pull request by bpaunescu**

### CODE REVIEW

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/cryptoservice/SupportedCryptoServices.kt [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
+
 enum class SupportedCryptoServices {
     /** Identifier for [BCCryptoService]. */
-    BC_SIMPLE
+    BC_SIMPLE;
     // UTIMACO, // Utimaco HSM.
     // GEMALTO_LUNA, // Gemalto Luna HSM.
     // AZURE_KV // Azure key Vault.
+
+    companion object {
```
Comment (dimosr) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/cryptoservice/SupportedCryptoServices.kt]: can we host this logic in a new component, i.e. `CryptoServiceFactory` ?

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/cryptoservice/SupportedCryptoServices.kt [hunk 2]
```diff
@@ -1,9 +1,24 @@
...
+        fun makeCryptoService(legalName: CordaX500Name, signingCertificateStore: FileBasedCertificateStoreSupplier? = null, cryptoServiceName: SupportedCryptoServices? = null, cryptoServiceConf: Path? = null): CryptoService {
```
Comment (dimosr) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/cryptoservice/SupportedCryptoServices.kt]: Making `cryptoServiceName` mandatory at this level could make this API a bit cleaner.

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/cryptoservice/SupportedCryptoServices.kt (hunk 1 above)
Comment (bpaunescu) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/cryptoservice/SupportedCryptoServices.kt]: done

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/cryptoservice/SupportedCryptoServices.kt (hunk 2 above)
Comment (bpaunescu) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/cryptoservice/SupportedCryptoServices.kt]: Done

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/cryptoservice/SupportedCryptoServices.kt [hunk 3]
```diff
@@ -1,5 +1,7 @@
 package net.corda.nodeapi.internal.cryptoservice
 
+import net.corda.nodeapi.internal.cryptoservice.bouncycastle.BCCryptoService
```
Comment (dimosr) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/cryptoservice/SupportedCryptoServices.kt]: remove please

**Pull request merged**